src/Generation/GirTool/GenerateCommand.cs

```diff
@@ -137,15 +137,14 @@ private static (DeserializedInput, DeserializedInput, DeserializedInput) LoadRep
         return (linuxRepositories!, macosRepositories!, windowsRepositories!);
     }
 
-    private static DeserializedInput DeserializeInput(string? searchPath, string[] input)
+    private static DeserializedInput DeserializeInput(string platformName, string? searchPath, string[] input)
     {
-        if (searchPath is null)
-            return DeserializedInput.Empty();
+        var repositoryResolver = GetRepositoryResolver(platformName, searchPath);
 
         var inputRepositories = input
-            .Select(x => Path.Join(searchPath, x))
-            .Where(File.Exists)
-            .Select(x => new FileInfo(x).OpenRead().DeserializeGirInputModel())
+            .Select(fileName => repositoryResolver.ResolveRepository(fileName))
+            .Where(inputRepository => inputRepository != null)
```

Comment:
```suggestion
            ..OfType<GirLoader.Input.Repository>()
```

This automatically discards all `null` values.

Reply:
I'm not sure I 100% agree about this being easier to understand, it's less obvious to me that the intent of this is to filter out null values, but I do think this is nicer on balance due to not needing to extra select to tell the compiler that the values are non-null.